Library/Homebrew/utils/formatter.rb

```diff
@@ -29,7 +29,12 @@ def success(string, label: nil)
     label(label, string, :green)
   end
 
+  # Ignores duplicate warning messages
   def warning(string, label: nil)
+    @warning_cache ||= Set.new
+    return if @warning_cache.include?(string)
```

Comment:
Looks like this only displays the warning only once even if the warning is triggered a second time for a different reason -- do we want that behaviour?

Reply:
That's a good question. I assumed that duplicate warning messages wouldn't be useful but I'll look over the ones already in the code base just to make sure. In this case, each error message had the formula name in it but there might be some more general ones we'd like to show multiple times. Do you know of any examples of those off the top of your head?

Another alternative would be to store both the error message and calling location but that might be going overboard.

Reply:
After double checking this doesn't look like something we'd want. There are a minority of warning messages that would make sense to show multiple times because they could get shown for each formula argument for example. There doesn't seem to be a good way to differentiate between them in this case so I'm gonna close this.